crates/wasmparser/src/parser.rs

```diff
@@ -17,14 +20,14 @@ use std::iter;
 /// This primary function for a parser is the [`Parser::parse`] function which
 /// will incrementally consume input. You can also use the [`Parser::parse_all`]
 /// function to parse a module that is entirely resident in memory.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Copy)]
```

Comment:
Although this is indeed `Copy` under the hood it's a stateful type where subtle bugs can happen due to where rustc may emit copies or not. In the same sense that iterators are generally not supposed to be `Copy` I think it might be best for this to remain not-`Copy`, although I'm not 100% sure why this was added so I could be convinced otherwise!

Reply:
I'll remove Copy from this and the component parser.